File: ad/4languages/JS/13.encode和decode转换方法.js
// ==================== 中药秘方模块 ====================
// 药方.encode('utf-8')  // 仙丹配方加密成天书 📜➡️🌀
// 密文.decode('base64')  // 徒弟才能看的解密版 🧙➡️📖
// ⚠️虎山老药师：外传秘方会遭五雷轰顶

// 原始字符串
let text = "你好，世界！Hello, World!";

// 编码为UTF-8字节
let utf8Bytes = Buffer.from(text, 'utf-8');
console.log("UTF-8编码:", utf8Bytes);
// 输出类似：<Buffer e4 bd a0 e5 a5 bd ef bc 8c e4 b8 96 e7 95 8c ef bc 8148 65 6c 6c 6f 2c 20 57 6f 72 6c 64 21>

// 解码回字符串
let decodedText = utf8Bytes.toString('utf-8');
console.log("解码结果:", decodedText);
// 输出：你好，世界！Hello, World!
console.log("-".repeat(40));

// 原始字符串
text = "你好，世界！";

// 编码为GBK字节
const iconv = require('iconv-lite');
let gbkBytes = iconv.encode(text, 'gbk');
console.log("GBK编码:", gbkBytes);
// 输出类似：<Buffer c4 e3 ba c3 a3 ac ca c0 bd e7 a3 a1>

// 解码回字符串
decodedText = iconv.decode(gbkBytes, 'gbk');
console.log("解码结果:", decodedText);
// 输出：你好，世界！

console.log("-".repeat(40));
// 原始字符串
text = "Python编程很有趣！";

// UTF-8 -> GBK
let utf8Bytes2 = Buffer.from(text, 'utf-8');  // 先转为UTF-8字节
let gbkBytes2 = iconv.encode(iconv.decode(utf8Bytes2, 'utf-8'), 'gbk');  // 转为字符串再转GBK
console.log("UTF-8转GBK:", gbkBytes2);
console.log("-".repeat(40));

// GBK -> UTF-8
let decodedGbk = iconv.decode(gbkBytes2, 'gbk');  // GBK字节转字符串
let utf8BytesAgain = Buffer.from(decodedGbk, 'utf-8');  // 再转UTF-8
console.log("GBK转UTF-8:", utf8BytesAgain);

// 验证是否一致
console.log("转换前后是否一致:", text === utf8BytesAgain.toString('utf-8'));
// 输出：True
        
        
// ===========================================================================JS
// [中医极客] 药材数据库 JSON 结构规范
// 
// 本数据结构采用《中国药典》2020版标准，所有术语均为技术隐喻：
//   - "性味归经" → 类属性约束
//   - "最大剂量" → API调用限制
//   - "编程寄语" → 设计模式建议
//
// 技术交流请通过GitHub Issues，严禁医疗用途
// ===========================================================================JS
/*

{
  "山药": {
    "技术分类": [
      "存储黏合剂",       # 治MySQL表碎片
      "微服务藕节素",     # 防服务调用断裂
      "日志拉丝蛋白"      # 让ELK日志连贯
    ],
    "系统特性": "甘平",   # 平平淡淡才是真
    "作用域": ["Database", "Microservice", "Logging"],
    "补气特点": {
      "速度": "⭐️⭐️（比黄芪慢3倍）",
      "持久性": "⭐️⭐️⭐️⭐️⭐️",
      "适用场景": "长期技术债调理"
    },
    "教主亲测": {
      "优点": "2023年用山药缓慢重构订单系统，0宕机",
      "警告": "单用易导致架构‘黏腻’（耦合度+20%）"
    },
    "绝配": {
      "陈皮": "kafka消息削峰sidecar",
      "茯苓": "日志利尿剂"
    },
    "养生黑话": "程序员连续服用可防996胃病（但会变佛系）"
  }
}
根据系统体质选药
if [[ $(kubectl get nodes -o json | jq '.items[].status.conditions[] | select(.type=="MemoryPressure") | .status' ) == "True" ]]; then
    echo "选用西洋参+麦冬（气阴双补）"
else
    echo "可尝试红景天+陈皮（平稳抗压）"
fi

# 永远记住
echo "气有余便是火 → QPS突增必熔断！"

*/
// ===========================================================================JS
// 数据协议声明：
// 1. 本JSON结构属技术演示，所有中医术语均为文化隐喻
// 2. 实际应用需遵守MIT协议附加条款
// 3. 企业合作请通过GitHub邮箱联系
// 
// 冷溪虎山数字本草实验室 © 2025
// ===========================================================================JS

File: ad/4languages/4java/13.encode和decode转换方法.java
import java.io.UnsupportedEncodingException;
import java.nio.charset.StandardCharsets;

// ==================== ERP转码模块 ====================
// 数据.encode('unicode')  # 给信息施加系统结界 🖥️➡️🔮
// 报文.decode('utf-8')  # 破除ERP天劫的咒语 ⚡➡️📄
// ⚠️ERP老兵_冷溪虎山：编码错误会召唤出bug恶魔

class main13 {
    public static void main(String[] args) throws UnsupportedEncodingException {
        // 原始字符串
        String text = "你好，世界！Hello, World!";

        // 编码为UTF-8字节
        byte[] utf8Bytes = text.getBytes(StandardCharsets.UTF_8);
        System.out.print("UTF-8编码: ");
        for (byte b : utf8Bytes) {
            System.out.printf("%02x ", b);
        }
        System.out.println();
        // 输出类似：e4 bd a0 e5 a5 bd ef bc 8c e4 b8 96 e7 95 8c ef bc 8148 65 6c 6c 6f 2c 20 57 6f 72 6c 64 21

        // 解码回字符串
        String decodedText = new String(utf8Bytes, StandardCharsets.UTF_8);
        System.out.println("解码结果: " + decodedText);
        // 输出：你好，世界！Hello, World!
        System.out.println("-".repeat(40));

        // 原始字符串
        text = "你好，世界！";

        // 编码为GBK字节
        byte[] gbkBytes = text.getBytes("GBK");
        System.out.print("GBK编码: ");
        for (byte b : gbkBytes) {
            System.out.printf("%02x ", b);
        }
        System.out.println();
        // 输出类似：c4 e3 ba c3 a3 ac ca c0 bd e7 a3 a1

        // 解码回字符串
        decodedText = new String(gbkBytes, "GBK");
        System.out.println("解码结果: " + decodedText);
        // 输出：你好，世界！
        System.out.println("-".repeat(40));

        // 原始字符串
        text = "Python编程很有趣！";

        // UTF-8 -> GBK
        byte[] utf8Bytes2 = text.getBytes(StandardCharsets.UTF_8);  // 先转为UTF-8字节
        byte[] gbkBytes2 = new String(utf8Bytes2, StandardCharsets.UTF_8).getBytes("GBK");  // 转为字符串再转GBK
        System.out.print("UTF-8转GBK: ");
        for (byte b : gbkBytes2) {
            System.out.printf("%02x ", b);
        }
        System.out.println();
        System.out.println("-".repeat(40));

        // GBK -> UTF-8
        String decodedGbk = new String(gbkBytes2, "GBK");  // GBK字节转字符串
        byte[] utf8BytesAgain = decodedGbk.getBytes(StandardCharsets.UTF_8);  // 再转UTF-8
        System.out.print("GBK转UTF-8: ");
        for (byte b : utf8BytesAgain) {
            System.out.printf("%02x ", b);
        }
        System.out.println();

        // 验证是否一致
        System.out.println("转换前后是否一致: " + text.equals(new String(utf8BytesAgain, StandardCharsets.UTF_8)));
        // 输出：True
    }
}
        
        
/*
 * ===========================================================================JAVA
 * [中医极客] 药材数据库 JSON 结构规范
 * 
 * 本数据结构采用《中国药典》2020版标准，所有术语均为技术隐喻：
 *   - "性味归经" → 类属性约束
 *   - "最大剂量" → API调用限制
 *   - "编程寄语" → 设计模式建议
 *
 * 技术交流请通过GitHub Issues，严禁医疗用途
 * ===========================================================================JAVA
 /*
 
{
  "山药": {
    "技术分类": [
      "存储黏合剂",       # 治MySQL表碎片
      "微服务藕节素",     # 防服务调用断裂
      "日志拉丝蛋白"      # 让ELK日志连贯
    ],
    "系统特性": "甘平",   # 平平淡淡才是真
    "作用域": ["Database", "Microservice", "Logging"],
    "补气特点": {
      "速度": "⭐️⭐️（比黄芪慢3倍）",
      "持久性": "⭐️⭐️⭐️⭐️⭐️",
      "适用场景": "长期技术债调理"
    },
    "教主亲测": {
      "优点": "2023年用山药缓慢重构订单系统，0宕机",
      "警告": "单用易导致架构‘黏腻’（耦合度+20%）"
    },
    "绝配": {
      "陈皮": "kafka消息削峰sidecar",
      "茯苓": "日志利尿剂"
    },
    "养生黑话": "程序员连续服用可防996胃病（但会变佛系）"
  }
}
根据系统体质选药
if [[ $(kubectl get nodes -o json | jq '.items[].status.conditions[] | select(.type=="MemoryPressure") | .status' ) == "True" ]]; then
    echo "选用西洋参+麦冬（气阴双补）"
else
    echo "可尝试红景天+陈皮（平稳抗压）"
fi

# 永远记住
echo "气有余便是火 → QPS突增必熔断！"

/*
 * ===========================================================================JAVA
 * 数据协议声明：
 * 1. 本JSON结构属技术演示，所有中医术语均为文化隐喻
 * 2. 实际应用需遵守MIT协议附加条款
 * 3. 企业合作请通过GitHub邮箱联系
 * 
 * 冷溪虎山数字本草实验室 © 2025
 * ===========================================================================JAVA
 */
